### DISCUSSION TITLE
Codemagic is not sending "RUNNING" status to Gitlab

### DISCUSSION THREAD

**Discussion by PawlikMichal25** [codemagic-ci-cd/codemagic-docs#1912] [Ideas]
### Use case

```Text
Codemagic is correctly sending other statuses for pipeline updates like "SUCCESS", "FAILED" or even "CANCELED", but it is not sending update for "RUNNING" - when the pipeline starts.

This a problem, because without it we cannot use "Merge automatically when pipeline succeeds" on Gitlab. It's also confusing because we're unsure if the pipeline even started while on Gitlab.
```

### Idea

```Text
Well, add it :)  
Considering you already have working implementation for other statuses, adding this one should be relatively easy.
```

**Comment by Ycoi** [comment]
Thanks for the idea @PawlikMichal25 I’ll discuss this with the team and we’ll see what we can do.  

Just from completeness, are there also other statuses in gitlab that you use and would benefit from or is it currently only the ‘running’ status which you miss in your normal usage?

**Comment by PawlikMichal25** [comment]
For completeness also `pending` could be added, so that we get the status on Gitlab immediately (when pipeline is pending on Codemagic)

**Comment by mikhail-tokarev** [comment]
Hey!
we've created a poll https://twitter.com/codemagicio/status/1687098415183007744 
please help to prioritise this feature 🙏

**Comment by PawlikMichal25** [comment]
@mikhail-tokarev any timelines?

**Comment by Afur** [comment]
We're also facing this issue :/

**Comment by PawelPalacz1** [comment]
Any updates ?

**Comment by PatrykMatwijczuk** [comment]
We're dealing with this issue too. Any updates about it would be awesome.

**Comment by bouzlibop** [comment]
It would be great to have these statuses: `running`, `pending` 👍

**Comment by artemii-yanushevskyi** [comment]
We now send `running` status to GitLab when a build starts. Thanks for bringing it up!

**Comment by CuriousDev21** [comment]
Just saw this thread, could I ask about to make this work. My understanding was that it is not possible to link codemagic build pipelines with MRs. Would love to know more.

**Reply by dam-ease** [reply]
Hey @CuriousDev21,
Yes, sure, this is possible.
You'd need to set up Webhooks for Gitlab as described in this section of the docs: https://docs.codemagic.io/yaml-running-builds/webhooks/#setting-up-webhooks-for-gitlab. After successful configuration, you can navigate to your project's commits in Gitlab to view the status of your builds.
<details><summary>Screenshot</summary>
<p>

<img width="823" alt="Screenshot 2025-03-11 at 15 43 03" src="https://github.com/user-attachments/assets/a1c0cb16-80e3-4f77-a6e6-07a3303d9fa8" />


</p>
</details>

**Reply by CuriousDev21** [reply]
Thank you for your feedback. I tried and i still can"t see any codemagic pipeline status in gitlab. 
The webhook has the following rights:
![Screenshot 2025-03-11 at 17 04 03](https://github.com/user-attachments/assets/834902db-b90d-490d-b3d5-1d1592539723)

**Reply by dam-ease** [reply]
Hey,
Thanks for your response!
Is this a self-hosted Gitlab instance? If yes, currently, with self-hosted instances, it is not possible to show pipeline results directly. But you can make use of our build status API in your workflow, as shown [here](https://docs.codemagic.io/rest-api/builds/#get-build-status), and then use [GitLab API](https://docs.gitlab.com/ee/api/commits.html#set-the-pipeline-status-of-a-commit) to post the status.
If not, can you please set up triggers on [push](https://docs.codemagic.io/yaml-running-builds/starting-builds-automatically/#build-triggers) and confirm in your Codemagic UI > Webhooks that the webhook was received? 
<img width="625" alt="Screenshot 2025-03-12 at 09 45 34" src="https://github.com/user-attachments/assets/ff4b0851-c713-4bbd-b744-2c3a26bdd812" />

**Reply by CuriousDev21** [reply]
@dam-ease Thank you so much for the hint, I managed to make both APIs work well to notify the pipeline status on a self hosted gitlab instance.
